interfaces/builtin/pipewire_server.go

```diff
@@ -0,0 +1,151 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/seccomp"
+	"github.com/snapcore/snapd/interfaces/udev"
+	"github.com/snapcore/snapd/snap"
+)
+
+// The pipewire-server interface is the companion interface to the audio-record
+// interface. The design of this interface is based on the idea that the slot
+// implementation (eg pulseaudio) is expected to query snapd on if the
+// audio-record slot is connected or not and the audio service will mediate
+// recording (ie, the rules below allow connecting to the audio service, but do
+// not implement enforcement rules; it is up to the audio service to provide
+// enforcement). If other audio recording servers require different security
+// policy for record and playback (eg, a different socket path), then those
+// accesses will be added to this interface.
+
+const pipewireServerSummary = `allows full access to the pipewire socket (don't needed for normal apps)`
+
+const pipewireServerBaseDeclarationSlots = `
+  pipewire-server:
+    allow-installation:
+      slot-snap-type:
+        - app
+        - core
+    deny-connection:
+      on-classic: false
+    deny-auto-connection: true
+`
+
+const pipewireServerConnectedPlugAppArmor = `
```

Comment:
As commented, this isn't just for audio, but for the whole pipewire. The snapped applications get access through portals, it's true, but the portals are offered by xdg-desktop-portal application, which requires access to the pipewire socket. So if pipewire is running inside a snap, it must be able to create the socket in the "global" XDG_RUNTIME_DIR, where the external xdg-desktop-portal application expects to find it. Also, a future snapped xdg-desktop-portal should use this same interface to be able to access to the pipewire socket. Finally, Gnome Shell requires access to the pipewire socket to do screen sharing, and it expects to find it at the "global" XDG_RUNTIME_DIR folder (and also, a snapped Gnome Shell will require this interface to get access to the pipewire socket placed at the "global" XDG_RUNTIME_DIR).

Reply:
Also, it's not a good idea to give access to the pipewire socket to any snapped application, because, as commented, they already can access it using portals, which is The Right Way(tm) :-D